### ISSUE TITLE
It seems to just stop?

### ISSUE THREAD

**Issue by kkjdaniel** [zumwald/oss-attribution-generator#5]
I run the command in my project, it quickly lists all the files its processing and then suddenly stalls and doesn't appear to continue.

Any help?

**Comment by zumwald** [comment]
How long are you noticing it stall for? If you're project is large, it can take a bit of time to traverse  everything (minutes, in the case of an extremely large project I've seen).

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by kkjdaniel** [comment]
It appears to have stopped here and hasn't continued for about 10 minutes...

![screen shot 2017-07-27 at 5 50 16 pm](https://user-images.githubusercontent.com/7441747/28682297-37ec1832-72f4-11e7-841b-1d85dda0ce10.png)

EDIT: Going on 20 now, with no further output.

**Comment by zumwald** [comment]
@kkjdaniel can you share a gist with the symptomatic package.json?

**Comment by kkjdaniel** [comment]
@zumwald Here is a link... https://gist.github.com/kkjdaniel/acb9dd4c007b5f12bdceaab67d6564bc

**Comment by kkjdaniel** [comment]
@zumwald Any luck spotting what might be causing the issue?

**Comment by zumwald** [comment]
@kkjdaniel tried installing your package.json as-is and it errors due to your dependency on `fs@*` - is there a reason you're specifying a built-in module? My Node version is 6.10.3 and npm is at 3.10.10 (slightly lagging latest LTS, but not far off). I also noticed your dependencies contain a number of packages which appear to actually be devDependencies, it'd be helpful for you to keep those distinctions accurate for using this tool.

Here's the full repro steps in Powershell on Windows:
```
PS N:\repos> mkdir test


    Directory: N:\repos


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        8/16/2017   8:20 AM                test


PS N:\repos> cd .\test\
PS N:\repos\test> Get-Clipboard | Set-Content ./package.json
PS N:\repos\test> npm i
npm WARN deprecated retext-inspect@0.4.2: switch to unist-util-inspect for future development
npm ERR! Windows_NT 10.0.15063
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\dazumwal\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\
\npm-cli.js" "i"
npm ERR! node v6.10.3
npm ERR! npm  v3.10.10
npm ERR! code EREADFILE

npm ERR! Error extracting C:\Users\dazumwal\AppData\Roaming\npm-cache\fs\0.0.0\package.tgz archive: ENOENT: no such file
 or directory, open 'C:\Users\dazumwal\AppData\Roaming\npm-cache\fs\0.0.0\package.tgz'
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     N:\repos\test\npm-debug.log
PS N:\repos\test>
```

**Comment by Kyle-Falconer** [comment]
It's also hanging for me:

![screen shot 2017-09-21 at 11 57 13 am](https://user-images.githubusercontent.com/1034255/30713431-138c7af8-9ec4-11e7-991b-477a116f62d6.png)

And all I have in the package.json's dependencies is:

```
    "dependencies": {
        "chai": "^3.5.0",
        "react": "15.4.2",
        "react-native": "0.42.3"
    }
```

This is happening both on Node v6.10 and Node v8.5.0 (I used nvm to switch to see if that was causing it).

## Edit:
I tried isolating the package.json as mentioned in the previous comment and I can do the `npm i` step just fine, and when I do the `attribution-generator`, it still hangs at the same place. Moving `chai` to the `devDependencies` block makes no difference.

I trimmed my package.json file to the following, and the issue can still be reproduced:

```
{
    "name": "foo",
    "version": "1.0.0",
    "description": "foo",
    "main": "index.js",
    "author": "bar",
    "license": "ISC",
    "dependencies": {
        "react": "15.4.2"
    }
}
```

## Edit 2:
It seems that it does eventually finish, but it takes hours to finish (I let it run for 3 hours and it finally did finish).

**Comment by zumwald** [comment]
I've noticed this too, perf is unacceptable. This, along with proper testing and CI/CD are on my radar to tackle before the holidays. I recognize that without the latter two, contributions are high-friction which makes this issue all the more painful for you @Kyle-Falconer.

**Comment by pz325** [comment]
Just to follow up this thread:

Hangging for me at:

...
processing wordwrap@0.0.2
processing wordwrap@0.0.3
processing wordwrap@1.0.0
processing wrappy@1.0.2
processing xmlbuilder@8.2.2
processing xtend@4.0.1
processing yargs@3.10.0
processing yauzl@2.4.1

**Comment by zumwald** [comment]
fixed in 1.6.0 thanks to @kconner and @rohithjidagam!
